pipeline/container/frontend/src/components/play/chat-app/ChatApp.tsx

```diff
@@ -91,10 +87,26 @@ export default function ChatApp({ pipeline }: ChatAppProps): JSX.Element {
       title: "Saved chat settings",
     });
 
-    // Settings come in as first dict value
-    setChatSettings(inputs[0].value);
+    // Map the input value to the dynamic fields default value
+    const updatedSettingsDynamicFields: DynamicFieldData[] | undefined =
+      dynamicFields[0].dicts &&
+      dynamicFields[0].dicts.map((dict, index) => {
+        return {
+          ...dict,
+          defaultValue: inputs[index].value,
+        };
+      });
+
+    // Update the dynamic fields state
+    setDynamicFields((prev) => {
+      prev[0].dicts = updatedSettingsDynamicFields;
```

Comment:
Think we may be mutating state directly here. May be worth changing this, what do you think?